### PR TITLE
Shanoir URI generation

### PR DESCRIPTION
**Pull request by alaeessaki**
This PR contains new code for generating Shanoir URI. 
It contains also the external platform keycloak attributes extension, and it wouldn't create any conflict with any code of the previous PR.

### CODE REVIEW

File: vip-datamanagement/src/main/java/fr/insalyon/creatis/vip/datamanager/server/business/ShanoirStorageBusiness.java [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+            throws RuntimeException, BusinessException {
+        if ( ! externalPlatform.getType().equals(Type.SHANOIR)) {
+            logger.error("Trying to generate a shanoir URI for a non shanoir storage {}",
+                    externalPlatform.getType());
+            throw new BusinessException("Cannot generate shanoir uri");
+        }
+        if (externalPlatform.getUrl() == null) {
+            logger.error("A shanoir external storage must have an URL to generate an URI");
+            throw new BusinessException("Cannot generate shanoir uri");
+        }
```
Comment (axlbonnet) [inline vip-datamanagement/src/main/java/fr/insalyon/creatis/vip/datamanager/server/business/ShanoirStorageBusiness.java:98]: Maybe also verify `externalPlatform.getKeycloakClientId()` and `externalPlatform.getRefreshTokenUrl()` and `externalPlatform.getUploadUrl` ?

**Review by axlbonnet — COMMENTED** [review]
Good stuff, only a small suggestion.
However, there is a conflict now that the other PR is accepted, you should merge develop into this branch to resolve it.

**Comment by alaeessaki** [comment]
Hello Axel, 
Can you check the new updates.
Thanks

File: vip-api/src/main/java/fr/insalyon/creatis/vip/api/CarminProperties.java [hunk 2]
```diff
@@ -64,7 +64,7 @@ public interface CarminProperties {
     // CARMIN PROCESSING
 
     String API_PIPELINE_WHITE_LIST = "carmin.processing.pipelines.whitelist";
-    String ADDITIONNAL_INPUT_VALID_CHARS = "\\[\\]?&="; // will be use in regex, so brackets must be escaped
+    String ADDITIONNAL_INPUT_VALID_CHARS = "\\[\\]"; // will be use in regex, so brackets must be escaped
```
Comment (axlbonnet) [inline vip-api/src/main/java/fr/insalyon/creatis/vip/api/CarminProperties.java:67]: This has been removed, so the `?&=` are not necessary anymore ?

Comment (alaeessaki) [reply vip-api/src/main/java/fr/insalyon/creatis/vip/api/CarminProperties.java:67]: Ah, I didn't see that it's deleted, I will fix this. It's necessary to send a URI in parameters.

Comment (alaeessaki) [reply vip-api/src/main/java/fr/insalyon/creatis/vip/api/CarminProperties.java]: it's fixed, you can check now

**Pull request merged**